### PR TITLE
test: Skip docker login for running tests

### PR DESCRIPTION
**Pull request by neilkakkar**
## Problem

Trying to docker login prevents contributor PRs from running this required test

## Changes

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

### CODE REVIEW

**Review by xvello — COMMENTED** [review]
Without this, we hit docker hub unauthenticated rate-limits at peak hours. I am curious that we don't hit them in `ci-plugin-server` too?

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -91,13 +91,6 @@ jobs:
                   sparse-checkout: 'rust/'
                   sparse-checkout-cone-mode: false
 
-            - name: Login to DockerHub
-              if: needs.changes.outputs.rust == 'true'
-              uses: docker/login-action@v2
-              with:
-                  username: posthog
-                  password: ${{ secrets.DOCKERHUB_TOKEN }}
```
Comment (Twixes) [inline .github/workflows/rust.yml:99]: To fix the issue while still addressing @xvello rate limit consideration, let's keep the step, but with [`continue-on-error: true`](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepscontinue-on-error)

**Comment by neilkakkar** [comment]
interesting, thanks for the context Xavier, yeah I expect the CI for plugin-server runs a lot more often than CI for rust, so surprising its all fine there

**Review by Twixes — APPROVED** [review]
:shipit:

**Pull request merged**